notebooks/tetraopt_example_using_comms_channel.ipynb

```diff
@@ -0,0 +1,175 @@
+    "        org_sample = org_list[0]\n",
+    "\n",
+    "        proj_list = list_all_projects(client=client, organization_id=org_sample.id)\n",
+    "        proj_sample = proj_list[0]\n",
+    "\n",
+    "        exp_list = list_all_experiments(client=client, project_id=proj_sample.id)\n",
+    "        exp_sample = exp_list[0]\n",
+    "\n",
+    "        print(f\"Running experiment within: \\n Org: {org_sample.id} \\n Proj: {proj_sample.id} \\n Exp: {exp_sample.id}\")\n",
+    "\n",
+    "        # Create the communication channels which will receive/send the inputs and outputs from/to TetraOpt\n",
+    "        objective_func_channel = await Channel.create(client=client)\n",
+    "        local_opt_channel = await Channel.create(client=client)\n",
+    "        print(\"objective_func_channel id: \", objective_func_channel)\n",
+    "        print(\"local_opt_channel id: \", local_opt_channel)\n",
+    "\n",
+    "        tetra_opt_params = {\n",
+    "            \"dimensionality\": 2,\n",
+    "            \"iteration_number\": 2,\n",
+    "            \"maximal_rank\": 4,\n",
+    "            \"points_number\": 1,\n",
+    "            \"quantization\": False,\n",
+    "            \"tolerance\": 0.0010000000474974513,\n",
+    "            \"lower_limits\": [0, 0],\n",
+    "            \"upper_limits\": [9, 9],\n",
+    "            \"grid\": [10, 10],\n",
+    "            \"objective_function_channel_id\": objective_func_channel.id,\n",
+    "            \"local_optimizer_channel_id\": local_opt_channel.id,\n",
+    "        }\n",
+    "\n",
+    "        run = ExperimentRun.create(\n",
+    "            client=client,\n",
+    "            algorithm=AlgorithmProto.TETRA_OPT,\n",
+    "            experiment_id=exp_sample.id,\n",
+    "            compute=HardwareProto.SMALL,\n",
+    "            parameters={'parameters': tetra_opt_params, 'inputs': {}}\n",
+    "        )\n",
+    "\n",
+    "        print(f\"Experiment run id: {run.id}\")\n",
+    "\n",
+    "        await asyncio.gather(\n",
+    "            objective_func_channel.connect(\n",
+    "                callback=lambda ask: objective_function_callback(ask, objective_func_channel),\n",
+    "                finish_callback=success, max_duration_in_sec=None, message_timeout_in_sec=500\n",
+    "            ),\n",
+    "            local_opt_channel.connect(\n",
+    "                callback=lambda ask: local_optimization_function_callback(ask, local_opt_channel),\n",
+    "                finish_callback=success, max_duration_in_sec=None, message_timeout_in_sec=500\n",
+    "            )\n",
+    "        )\n",
+    "\n",
+    "        return run\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "59aea95b-1123-4120-8c8d-b7add6575901",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async def run_experiment():\n",
+    "    run = await main()\n",
+    "    print(run.poll())\n",
+    "\n",
+    "await run_experiment()"
```

Comment:
are you sure this is working? Does jupyter have a top level async support? 🤔

Reply:
yes, tested and working.